routes/documents.py

```diff
@@ -170,6 +171,8 @@ def submit_document():
     file = request.files['file']
     if not _allowed_file(file.filename):
         raise ClientError('file extension not allowed', status=406)
+    if data['validated'] and not current_user.is_authenticated:
```

Comment:
The "missing" parameter in fields.Boolean() should handle this. But the PR is outdated anyway :)